### PR TITLE
Use updated API to get the keys from the bucket

### PR DESCRIPTION
**Pull request by sharidas**
Use updated API like getPaginator instead of
getIterator to get the keys from the bucket.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

### CODE REVIEW

**Comment by sharidas** [comment]
#### Description

Due to the usage of old API `getIterator` there was a problem noticed when the core is upgraded to the latest version and with the usage of this app for aws s3 storage. The problem was the data inside the root folder of the storage in the View was not visible. And hence it is considered as data loss. Because subsequently the scanner was deleting the entries in the filecache table. This change helps to retreive the data.

#### Issue 
- Fixes https://github.com/owncloud/files_external_s3/issues/235

#### Tests done
- Install oC 10.2.1
- Now enable s3 storage from admin->settings->storage section
- Now add a folder and files inside the root folder of the storage
- Upgrade core to 10.3.0.alpha2 version. Before upgrading to 10.3.0, kindly install this app and enable it.
- Now after upgrade check the root folder and contents of the amazon s3 folder, it should be visible.

**Comment by sharidas** [comment]
Got to know about `getPaginator` from here https://docs.amazonaws.cn/en_us/sdk-for-php/v3/developer-guide/aws-sdk-php-v3-developer-guide.pdf ( refer: page 19 towards the bottom of the page )

File: lib/Storage/AmazonS3.php [hunk 1]
```diff
@@ -260,20 +260,18 @@ public function opendir($path) {
...
 				'Delimiter' => '/',
 				'Prefix' => $path
-			], ['return_prefixes' => true]);
+			]);
 
-			foreach ($result as $object) {
-				if (isset($object['Key']) && $object['Key'] === $path) {
-					// it's the directory itself, skip
+			foreach ($result->search('[CommonPrefixes[].Prefix, Contents[].Key][]') as $object) {
+				if (is_string($object) && $object === $path) {
```
Comment (jvillafanez) [inline lib/Storage/AmazonS3.php]: is the `is_string(..)` check needed?
if the `$object` isn't a string (assuming this is possible, I don't know) you'll be feeding the `basename` function with something which isn't a string.

Comment (sharidas) [reply lib/Storage/AmazonS3.php]: > if the $object isn't a string (assuming this is possible, I don't know) you'll be feeding the basename function with something which isn't a string.

I have updated the change. Now the value to `$file` fed from `basename` will be evaluated in the if condition with `is_string`. The reason why I have used `is_string` is, to be more defensive. If I get some different values, I am ignoring them.

**Comment by sharidas** [comment]
@jvillafanez I have moved the migration to this app.

**Pull request merged**